File: dayNine/python/parking_lot.py
class ParkingLot:

    SIZE = 20

    def __init__(self):

        self.slots = [0] * self.SIZE

    def park_next(self):

        for index in range(self.SIZE):

            if self.slots[index] == 0:
                self.slots[index] = 1
                return True

        return False

    def park_at(self, slot_number):

        index = slot_number - 1

        if index < 0 or index >= self.SIZE:
            return False

        if self.slots[index] == 1:
            return False


        self.slots[index] = 1

        return True

    def leave(self, slot_number):

        index = slot_number - 1

    
        if index < 0 or index >= self.SIZE:
            return False

        if self.slots[index] == 0:
            return False

        self.slots[index] = 0

        return True

    def get_available_spaces(self):

        count = 0

        for slot in self.slots:

            if slot == 0:
                count += 1

        return count

    def get_filled_spaces(self):

        count = 0

        for slot in self.slots:

            if slot == 1:
                count += 1

        return count

    def display_status(self):

        print("\n----- PARKING STATUS -----")

        for index in range(self.SIZE):

            print(
                "Slot", index + 1, ":", self.slots[i]
            )

        print(
            "Available spaces:",
            self.get_available_spaces()
        )

        print(
            "Filled spaces:",
            self.get_filled_spaces()
        )

        print("--------------------------")

File: dayNine/java/ParkingLot.java
//javac -cp "junit-platform-console-standalone-1.11.0.jar:out" -d out ParkingLotTest.java ParkingLot.java
//java -cp "junit-platform-console-standalone-1.11.0.jar:out" org.junit.platform.console.ConsoleLauncher --scan-class-path
public class ParkingLot {


    private static final int SIZE = 20;


    private int[] slots;


    public ParkingLot() {
        slots = new int[SIZE];
    }


    public boolean parkNext() {

        for (int index = 0; index < SIZE; index++) {

            if (slots[index] == 0) {
                slots[index] = 1;
                return true;
            }
        }

        return false;
    }


    public boolean parkAt(int slotNumber) {


        int index = slotNumber - 1;

    
        if (index < 0 || index >= SIZE) {
            return false;
        }


        if (slots[index] == 1) {
            return false;
        }


        slots[index] = 1;

        return true;
    }


    public boolean leave(int slotNumber) {

        int index = slotNumber - 1;


        if (index < 0 || index >= SIZE) {
            return false;
        }


        if (slots[index] == 0) {
            return false;
        }

      
        slots[index] = 0;

        return true;
    }


    public int getAvailableSpaces() {

        int count = 0;

        for (int slot : slots) {
            if (slot == 0) {
                count++;
            }
        }

        return count;
    }

 
    public int getFilledSpaces() {

        int count = 0;

        for (int slot : slots) {
            if (slot == 1) {
                count++;
            }
        }

        return count;
    }

 
    public int[] getSlots() {
        return slots.clone();
    }

 
    public void displayStatus() {

        System.out.println("\n----- PARKING STATUS -----");

        for (int index = 0; index < SIZE; index++) {

            System.out.println(
                "Slot " + (index + 1) + ": " + slots[index]
            );
        }

        System.out.println(
            "Available spaces: " + getAvailableSpaces()
        );

        System.out.println(
            "Filled spaces: " + getFilledSpaces()
        );

        System.out.println("--------------------------");
    }
}
